### PR TITLE
implements GrantLease method for operationLeaseStorage port

### PR DESCRIPTION
**Pull request by caiordjesus**
### What was done?
- Implemented method GrantLease for operation_lease_storage port and unit tests for it.
### Why?
- To be able to understand which operations are running, and which expired, we first grant a lease for the specific operation.
### How?
- Using redis sorted set. We add a score (unixtime) with the information of when the operation expires.

### CODE REVIEW

File: internal/adapters/operation_lease/redis/redis.go [hunk 1]
```diff
@@ -64,3 +86,15 @@ func (r *redisOperationLeaseStorage) FetchLeaseTTL(ctx context.Context, schedule
 func (r *redisOperationLeaseStorage) ListExpiredLeases(ctx context.Context, schedulerName string, maxLease time.Time) ([]operation.OperationLease, error) {
 	return nil, nil
 }
+
+func (r *redisOperationLeaseStorage) buildSchedulerOperationLeaseKey(schedulerName string) string {
+	return fmt.Sprintf("operations:%s:operationsLease", schedulerName)
+}
+
+func (r *redisOperationLeaseStorage) existsOperationLease(ctx context.Context, schedulerName, operationId string) (bool, error) {
+	operationLeaseList, _, err := r.client.ZScan(ctx, r.buildSchedulerOperationLeaseKey(schedulerName), 0, operationId, 0).Result()
```
Comment (GuilhermBrSp) [inline internal/adapters/operation_lease/redis/redis.go:95]: @caiordjesus WDYT about using ZSCORE for checking the existence here? it's time complexity is constant in comparison with Zscan that have a linear complexity in worst case. Ref: https://redis.io/commands/zscore

Comment (GuilhermBrSp) [reply internal/adapters/operation_lease/redis/redis.go:95]:
```
operationLeaseScore, err := client.ZScore(ctx, r.buildSchedulerOperationLeaseKey(schedulerName), operationID).Result()
```

Comment (GuilhermBrSp) [reply internal/adapters/operation_lease/redis/redis.go:95]: se o erro for diferente de nil, significa que o valor que estamos buscando não existe ainda, então você poderia retornar apenas err != nil

File: internal/adapters/operation_lease/redis/redis.go [hunk 2]
```diff
@@ -64,3 +86,15 @@ func (r *redisOperationLeaseStorage) FetchLeaseTTL(ctx context.Context, schedule
...
+	if err != nil {
+		return false, errors.NewErrUnexpected("failed to list active operations for \"%s\"", schedulerName).WithError(err)
```
Comment (GuilhermBrSp) [inline internal/adapters/operation_lease/redis/redis.go:97]: This message should inform about operation lease and not active operations, right?

File: internal/adapters/operation_lease/redis/redis.go (hunk 1 above)
Comment (caiordjesus) [reply internal/adapters/operation_lease/redis/redis.go]: Awesome insight!!
Just made the change!

File: internal/adapters/operation_lease/redis/redis.go (hunk 2 above)
Comment (caiordjesus) [reply internal/adapters/operation_lease/redis/redis.go]: You are right! Just made the change! Thanks!!

**Review by GuilhermBrSp — APPROVED** [review]

**Pull request merged**